flink-ml-lib/src/main/java/org/apache/flink/ml/feature/vectorslicer/VectorSlicerParams.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.feature.vectorslicer;
+
+import org.apache.flink.ml.common.param.HasInputCol;
+import org.apache.flink.ml.common.param.HasOutputCol;
+import org.apache.flink.ml.param.IntArrayParam;
+import org.apache.flink.ml.param.Param;
+import org.apache.flink.ml.param.ParamValidators;
+
+/**
+ * Params of {@link VectorSlicer}.
+ *
+ * @param <T> The class type of this instance.
+ */
+public interface VectorSlicerParams<T> extends HasInputCol<T>, HasOutputCol<T> {
+    Param<Integer[]> INDICES =
+            new IntArrayParam(
+                    "indices",
+                    "An array of indices to select features from a vector column.",
+                    null,
+                    ParamValidators.numericalArrayGtEq(0));
```

Comment:
I just found that in order to align the behavior of this algorithm with existing practices, we need to make sure that the input indices are not duplicated. Could you please improve the param validator with this function, add JavaDoc to explain the behavior when there are duplicated input indices, and add corresponding tests?

Given that we need to verify the input integers are non-negative and non-duplicate, do you think it would be better to add a unique private param validator in this class to verify this specified case, instead of trying to add a common validator?

Reply:
I think one validator can not do all the checks of this parameter. We just do some checks. The duplicated check may do in the algorithm code. The common validator only do one kind of checks. We can do the other checks in some other place, just like the begin of the algorithm. 

What do you think about.

Reply:
I prefer to add an exclusive validator for VectorSlicer, as that is what spark `VectorSlicer.validIndices` has done.

If we add `checkIndices()`, it means that exceptions would be thrown in `transform()` rather than `setIndices()`. I think the latter one has better readability.

Reply:
+1 to add a validator for VectorSlicer. If we agree to do this, we probably should also remove the numericalArrayGtEq since it is not needed any more then.

Reply:
OK